include/okapi/api/chassis/controller/odomChassisController.hpp

```diff
@@ -0,0 +1,127 @@
+/**
+ * @author Ryan Benasutti, WPI
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#pragma once
+
+#include "okapi/api/chassis/controller/chassisController.hpp"
+#include "okapi/api/chassis/model/skidSteerModel.hpp"
+#include "okapi/api/coreProsAPI.hpp"
+#include "okapi/api/odometry/odometry.hpp"
+#include "okapi/api/util/timeUtil.hpp"
+
+namespace okapi {
+class OdomChassisController : public ChassisController {
+  public:
+  /**
+   * Odometry based chassis controller. Starts task at the default for odometry when constructed.
+   *
+   * Moves the robot around in the odom frame. Instead of telling the robot to drive forward or
+   * turn some amount, you instead tell it to drive to a specific point on the field or turn to
+   * a specific angle relative to its starting position.
+   *
+   * @param itimeUtil The TimeUtil.
+   * @param iparams odometry parameters for the internal odometry math
+   * @param imoveThreshold minimum length movement (smaller movements will be skipped)
+   * @param iturnThreshold minimum angle turn (smaller turns will be skipped)
+   */
+  OdomChassisController(const TimeUtil &itimeUtil,
+                        std::unique_ptr<Odometry> iodometry,
+                        const QLength &imoveThreshold = 10_mm,
+                        const QAngle &iturnThreshold = 1_deg);
+
+  ~OdomChassisController() override;
+
+  OdomChassisController(const OdomChassisController &) = delete;
+  OdomChassisController(OdomChassisController &&other) = delete;
+  OdomChassisController &operator=(const OdomChassisController &other) = delete;
+  OdomChassisController &operator=(OdomChassisController &&other) = delete;
+
+  /**
+   * Drives the robot straight to a point in the odom frame.
+   *
+   * @param ix x coordinate
+   * @param iy y coordinate
+   * @param ibackwards whether to drive to the target point backwards
+   * @param ioffset offset from target point in the direction pointing towards the robot
+   */
+  virtual void driveToPoint(const QLength &ix,
+                            const QLength &iy,
```

Comment:
What `StateMode` do these values refer to? It might be a good idea to have a flag in the chassis controller and maybe a parameter in the ctor to specify what mode the entire chassis uses.

Reply:
That parameter should be added, this is an oversight.